pkg/cluster/condition.go

```diff
@@ -28,65 +32,85 @@ func (m *manager) bootstrapConfigMapReady(ctx context.Context) (bool, error) {
 	return err == nil && cm.Data["status"] == "complete", nil
 }
 
-func (m *manager) apiServersReady(ctx context.Context) (bool, error) {
+func (m *manager) apiServersReady(ctx context.Context) (bool, bool, error) {
 	apiserver, err := m.configcli.ConfigV1().ClusterOperators().Get(ctx, "kube-apiserver", metav1.GetOptions{})
 	if err != nil {
-		return false, nil
+		return false, true, nil
 	}
-	return isOperatorAvailable(apiserver), nil
+	return isOperatorAvailable(apiserver), true, nil
 }
 
-func (m *manager) minimumWorkerNodesReady(ctx context.Context) (bool, error) {
+func (m *manager) minimumWorkerNodesReady(ctx context.Context) (bool, bool, error) {
 	nodes, err := m.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
 		LabelSelector: "node-role.kubernetes.io/worker",
 	})
+	//err = errors.New("ERROR FOR MIN WORKER NODES")
 	if err != nil {
-		return false, nil
+		return false, true, api.NewCloudError(
+			http.StatusInternalServerError,
+			api.CloudErrorCodeDeploymentFailed,
+			"", err.Error()+" Minimum number of worker nodes have not been successfully created. Please retry, and if the issue persists, raise an Azure support ticket",
+		)
 	}
 
 	readyWorkers := 0
+	message := ""
 	for _, node := range nodes.Items {
 		for _, cond := range node.Status.Conditions {
 			if cond.Type == corev1.NodeReady && cond.Status == corev1.ConditionTrue {
 				readyWorkers++
+			} else {
+				messageString := fmt.Sprintf("%+v - Status:%+v, Message: %+v\n", node, cond.Status, cond.Message)
+				message += messageString
 			}
 		}
 	}
 
-	return readyWorkers >= minimumWorkerNodes, nil
+	if readyWorkers >= minimumWorkerNodes {
+		return readyWorkers >= minimumWorkerNodes, false, nil
+	} else {
+		if message == "" {
+			message = "Check the config and versions"
+		}
+		return false, true, api.NewCloudError(
+			http.StatusInternalServerError,
+			api.CloudErrorCodeDeploymentFailed,
+			"", message+" Minimum number of worker nodes have not been successfully created. Please retry, and if the issue persists, raise an Azure support ticket",
```

Comment:
The string " Minimum number of worker nodes ... ticket" could be extracted to a package variable (or at the beginning of this function) as it is long and used twice.